### PR TITLE
[SPARK-44626][SS][CONNECT] Followup on streaming query termination when client session is timed out for Spark Connect

### PR DESCRIPTION
**Pull request by bogao007**
### What changes were proposed in this pull request?

Removed keep alive feature in `SparkConnectStreamingQueryCache` so that the session mapping of running queries can expire after a client session times out. This is needed for a change to terminate the query when client session is timed out.

### Why are the changes needed?

This is a followup of https://issues.apache.org/jira/browse/SPARK-44432. Without removing the keep alive feature, the session mapping for running queries cannot expire correctly since it would be accessed periodically

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Did manual testing and verified the change is working as expected. Adding unit test is kind of hard since the session mapping logic cannot be easily mocked. We want to get this checked in with Spark 3.5 asap and this is already verified with a manual testing.

Create a streaming query on client side and then disconnect:
```
@ val q = spark.readStream.format("rate").load().writeStream.format("console").start()

@ exit
```

Validate on server side that the running query is stopped after a while:
```
23/08/01 13:26:34 INFO SparkConnectStreamingQueryCache: Stopping the query with id 787cdaed-0e6a-4496-a4a3-78b77789193b since the session has timed out
23/08/01 13:26:34 INFO DAGScheduler: Asked to cancel job group c615e5a7-78ff-462b-b24c-bba55259f8f5
23/08/01 13:26:34 INFO MicroBatchExecution: Async log purge executor pool for query [id = 787cdaed-0e6a-4496-a4a3-78b77789193b, runId = c615e5a7-78ff-462b-b24c-bba55259f8f5] has been shutdown
23/08/01 13:26:34 INFO MicroBatchExecution: Deleting checkpoint file:/private/var/folders/b0/f9jmmrrx5js7xsswxyf58nwr0000gp/T/temporary-05b9e890-ec53-45df-a3ca-65c4c01a143d.
23/08/01 13:26:34 INFO DAGScheduler: Asked to cancel job group c615e5a7-78ff-462b-b24c-bba55259f8f5
23/08/01 13:26:34 INFO MicroBatchExecution: Query [id = 787cdaed-0e6a-4496-a4a3-78b77789193b, runId = c615e5a7-78ff-462b-b24c-bba55259f8f5] was stopped
```

### CODE REVIEW

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamingQueryCache.scala [hunk 1]
```diff
@@ -182,12 +179,10 @@ private[connect] class SparkConnectStreamingQueryCache(
             log.info(s"Removing references for $id in session ${v.sessionId} after expiry period")
             queryCache.remove(k)
 
-          case Some(_) => // Inactive query waiting for expiration. Keep the session alive.
-            sessionsToKeepAlive.add((v.userId, v.sessionId))
+          case Some(_) => // Inactive query waiting for expiration. Do nothing.
+            log.info(s"Waiting for the expiration for $id in session ${v.sessionId}")
```
Comment (WweiL) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamingQueryCache.scala:183]: This should be logInfo, also I think all other logs in this file should be logInfo.

**Review by WweiL — APPROVED** [review]

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamingQueryCache.scala [hunk 2]
```diff
@@ -168,9 +168,6 @@ private[connect] class SparkConnectStreamingQueryCache(
    */
   private def periodicMaintenance(): Unit = {
```
[the remark this answers is not in the document]
Comment (bogao007) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectStreamingQueryCache.scala:167]: Good catch! Updated

**Comment by bogao007** [comment]
Hi @ueshin @HyukjinKwon , could you help review this PR and merge it? It also needs to go to 3.5 branch, thanks!

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.

**Pull request closed without merging**